### PR TITLE
[MM-24957] Adding circle ci configuration and build scripts.

### PR DESCRIPTION
**Pull request by stylianosrigas**
#### Summary
This PR adds:
- Circle CI configuration for builds and testing
- Build scripts to build app and create docker image

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-24957

#### Release Note

```release-notes
NONE
```

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    - checkout
+    - *restore_cache
+    - run: |
+        go get golang.org/x/tools/cmd/cover
+        go get github.com/mattn/goveralls
+    - run: go test ./... -v -covermode=count -coverprofile=coverage.out
+    - *save_cache
+    - run:
+        name: coveralls
+        command: goveralls -coverprofile=coverage.out -service=circle-ci -repotoken $COVERALLS_REPO_TOKEN || true
```
Comment (cpanato) [inline .circleci/config.yml:40]: did you enable this?

File: build/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# Build the mattermost cloud
+ARG DOCKER_BUILD_IMAGE=golang:1.14.1
```
Comment (cpanato) [inline build/Dockerfile:2]: can update this
```suggestion
ARG DOCKER_BUILD_IMAGE=golang:1.14.2
```

File: .circleci/config.yml (hunk 1 above)
Comment (stylianosrigas) [reply .circleci/config.yml]: Not needed atm. I am removing it ;)

**Pull request merged**